account.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"math"
+	"time"
+
+	"github.com/cockroachdb/apd/v2"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	auth "github.com/cosmos/cosmos-sdk/x/auth/types"
+	vesting "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	bank "github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// Account is an internal representation of a genesis regen account
+type Account struct {
+	Address       sdk.AccAddress
+	TotalRegen    apd.Decimal
+	StartTime     time.Time
+	Distributions []Distribution
+	EndTime       time.Time
+}
+
+// Distribution is an internal representation of a genesis vesting distribution of regen
+type Distribution struct {
+	Time  time.Time
+	Regen apd.Decimal
+}
+
+const (
+	URegenDenom = "uregen"
+)
+
+var tenE6 *apd.Decimal
+
+func init() {
+	var err error
+	tenE6, _, err = apd.NewFromString("1000000")
+	if err != nil {
+		panic(err)
+	}
+}
+
+func ToCosmosAccount(acc Account) (auth.AccountI, *bank.Balance, error) {
+	totalCoins, err := RegenToCoins(&acc.TotalRegen)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	addrStr := acc.Address.String()
+	balance := &bank.Balance{
+		Address: addrStr,
+		Coins:   totalCoins,
+	}
+	if len(acc.Distributions) == 0 {
```

Comment:
On the other hand, if we wanted to save on code complexity, it might be just easier to encode all coins from accounts with `len(acc.Distributions) > 1` as vesting coints, and just have vesting time=0 for the first drop. While it seems a bit funky, i may be in favor of it if it makes our logic here more clean & readable.

Reply:
I'm dealing with this stuff in a separate phase - just haven't opened that PR yet. All the original logic is still in #1